src/main/java/com/everis/control/CarService.java

```diff
@@ -15,17 +18,23 @@
 
 	@Inject
 	private PersistenceService<Car, String> persistenceService;
+	
+	private Map<String, String> filterMap = new HashMap<String, String>();
 
 	/**
 	 * Method to get a list of Car Entity available
 	 */
 
-	public List<CarDto> getCars() {
-		List<CarDto> cars=this.persistenceService.getEntitiesWithNamedQuery("Car.findAll", Car.class).stream().
-				map(car -> car.mapToDto()).collect(Collectors.toList());
-		return cars;
+	public List<CarDto> getCars(int page, int size, String filterBy, String orderBy) {
+		
+		filterMap.put("brand", filterBy);
```

Comment:
I got you point about the filterMap, but I think if the user wants to filter he probably wants to filter by a specific field. Imagine that the user wants to filter by brand, he is not interested in which country or registration the car have.
Could you please double check the filtering logic ?